app/lib/application_form_status_updater.rb

```diff
@@ -88,19 +88,29 @@ def subjects_status
   end
 
   def work_history_status
-    return :not_started if has_work_history.nil?
-
-    if !has_work_history ||
-         (
-           !work_histories.empty? &&
-             work_histories.all? do |work_history|
-               work_history_complete?(work_history)
-             end
-         )
-      return :completed
-    end
+    all_work_histories_complete =
+      work_histories.all? do |work_history|
+        work_history_complete?(work_history)
```

Comment:
Would this make more sense on the `WorkHistory` model? I realise it was already here.

Reply:
This was actually originally where we had it before implementing the `ApplicationFormStatusUpdater`, I think I prefer it in this class as we want to make sure we're only using the cached `work_history_status` field and not interrogating individual work histories, but I'm happy either way.